src/main/java/com/kalessil/phpStorm/phpInspectionsEA/inspectors/forEach/ForeachSourceInspector.java

```diff
@@ -92,60 +113,76 @@ private boolean isEnsuredByPyParentIf(@NotNull ForeachStatement foreach, @NotNul
                 return result;
             }
 
-            private void analyseContainer(@NotNull PsiElement container) {
+            private void analyseContainer(@NotNull final PsiElement container) {
```

Comment:
Parameters don't need to be final, only if has a purpose (but local variables are final for sure).

Reply:
While it really is not need, I think that is valid to make sure that this parameter is never modified.

Reply:
No need ATM